From 7257a7a24919bf1fde4bdb3c132700ebdcc47492 Mon Sep 17 00:00:00 2001
From: Duy Dang <duy.dang.yb@renesas.com>
Date: Wed, 22 Jul 2020 11:51:00 +0700
Subject: [PATCH 277/476] ARM: dts: r8a7742: Add VCP/VPC nodes

Add Video Processing Unit Cache (VPC module) and
Multi-codec module (VCP module) to the R8A7742 (RZ/G1H) SoC dtsi.

These modules are for Renesas UVCS kernel driver module probe
(uvcs-kernel-module).

Signed-off-by: Duy Dang <duy.dang.yb@renesas.com>
---
 arch/arm/boot/dts/r8a7742.dtsi | 58 ++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 58 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index b8e1fb9..832f57a 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -1740,6 +1740,64 @@
 			status = "disabled";
 		};
 
+		vcp0: vcp0@fe900000 {
+			compatible = "renesas,vcp0";
+			reg = <0 0xfe900000 0 0x200>, <0 0xfe900200 0 0x200>;
+			interrupts = <0 258 IRQ_TYPE_LEVEL_HIGH>,
+				 <0 259 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A7742_CLK_VCP0>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+			renesas,#ch = <0>;
+		};
+
+		vpc0: vcp0@fe908000 {
+			compatible = "renesas,vpc0";
+			reg = <0 0xfe908000 0 0x90>;
+			clocks = <&mstp1_clks R8A7742_CLK_VPC0>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+			renesas,#ch = <0>;
+		};
+
+		vcp1: vcp1@fe900000 {
+			compatible = "renesas,vcp1";
+			reg = <0 0xfe910000 0 0x200>, <0 0xfe910200 0 0x200>;
+			interrupts = <0 260 IRQ_TYPE_LEVEL_HIGH>,
+				 <0 261 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A7742_CLK_VCP1>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+			renesas,#ch = <1>;
+		};
+
+		vpc1: vcp0@fe918000 {
+			compatible = "renesas,vpc1";
+			reg = <0 0xfe918000 0 0x90>;
+			clocks = <&mstp1_clks R8A7742_CLK_VPC1>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+			renesas,#ch = <1>;
+		};
+
+		vpc0xy: vcp0@fe960380 {
+			compatible = "renesas,vpc0xy";
+			reg = <0 0xfe960380 0 0x4>;
+			clocks = <&mstp1_clks R8A7742_CLK_VPC0>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+			renesas,#ch = <0>;
+		};
+
+		vpc1xy: vcp0@fe960384 {
+			compatible = "renesas,vpc1xy";
+			reg = <0 0xfe960384 0 0x4>;
+			#clocks = <&mstp1_clks R8A7742_CLK_VPC1>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+			renesas,#ch = <1>;
+		};
+
+		prr: vcp0@ff000044 {
+			compatible = "renesas,prr";
+			reg = <0 0xff000044 0 0x4>;
+			renesas,#ch = <0>;
+		};
+
 		vin0: video@e6ef0000 {
 			compatible = "renesas,vin-r8a7742",
 				     "renesas,rcar-gen2-vin";
-- 
2.7.4

